compiler/il/OMRBlock.cpp

```diff
@@ -1383,8 +1383,8 @@ OMR::Block::splitEdge(TR::Block *from, TR::Block *to, TR::Compilation *c, TR::Tr
    else
       {
       TR::TreeTop *lastTree = from->getLastRealTreeTop();
-      TR::ILOpCodes opCode = lastTree->getNode()->getOpCodeValue();
-      TR_ASSERT(opCode == TR::athrow || lastTree->getNode()->getOpCode().isReturn(), "edge to EXIT should end in throw or return");
+      TR::ILOpCode &opCode = lastTree->getNode()->getOpCode().isCheck() ? lastTree->getNode()->getFirstChild()->getOpCode() : lastTree->getNode()->getOpCode();
```

Comment:
Should we also check for TR_treetop opcode here ? I know that throw/return can appear at the top of the tree but some optimization could in theory change a null check to a treetop for example and wondering if it might be better to future proof this check.